src/types/tokens.ts

```diff
@@ -0,0 +1,55 @@
+import { SignableToken } from '../api';
+
+export interface ERC20Token {
+  type: 'ERC20';
+  tokenAddress: string;
+}
+
+export interface ERC721Token {
+  type: 'ERC721';
+  tokenId: string;
+  tokenAddress: string;
+}
+
+export interface ETHToken {
+  type: 'ETH';
+}
+
+export interface ETHAmount extends ETHToken {
+  amount: string;
+}
+export interface ERC20Amount extends ERC20Token {
+  amount: string;
+}
+
+export type AnyToken = ETHToken | ERC721Token | ERC20Token;
```

Comment:
It is unfortunate we cannot use `Token` here. I don't love `AnyToken` but I cannot come up with a better recommendation.

Reply:
Maybe it could be called `Asset`?

Reply:
Will clash with the existing `Assets` won't it?